File: gin-admin-backend/internal/data/dictionary.go
package data

import (
	"time"
)

type Dictionary struct {
	ID        string    `gorm:"primaryKey;type:varchar(36)" json:"id"`
	DictName  string    `gorm:"type:varchar(100)" json:"dictName"`
	DictCode  string    `gorm:"type:varchar(100);unique" json:"dictCode"`
	Status    int       `gorm:"type:tinyint;default:1" json:"status"`
	Remark    string    `gorm:"type:varchar(255)" json:"remark"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`

	Items []DictionaryItem `gorm:"foreignKey:DictID" json:"items"`
}

func (Dictionary) TableName() string {
	return "sys_dictionary"
}

type DictionaryItem struct {
	ID        string    `gorm:"primaryKey;type:varchar(36)" json:"id"`
	DictID    string    `gorm:"type:varchar(255);index" json:"dictId"`
	Label     string    `gorm:"type:varchar(100)" json:"label"`
	Value     string    `gorm:"type:varchar(100)" json:"value"`
	Sort      int       `gorm:"type:int;default:0" json:"sort"`
	Status    int       `gorm:"type:tinyint;default:1" json:"status"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

func (DictionaryItem) TableName() string {
	return "sys_dictionary_item"
}

File: python-admin-backend/app/models/dictionary.py
from sqlalchemy import Column, String, Integer, DateTime, Text, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.database import Base


class Dictionary(Base):
    __tablename__ = "sys_dictionary"

    id = Column(String(36), primary_key=True, index=True)
    dict_name = Column(String(100), nullable=False)
    dict_code = Column(String(100), unique=True, nullable=False)
    status = Column(Integer, server_default="1")
    remark = Column(Text, nullable=True)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    items = relationship("DictionaryItem", backref="dictionary", cascade="all, delete-orphan")


class DictionaryItem(Base):
    __tablename__ = "sys_dictionary_item"

    id = Column(String(36), primary_key=True, index=True)
    dict_id = Column(String(36), ForeignKey("sys_dictionary.id"), nullable=False)
    label = Column(String(100), nullable=False)
    value = Column(String(100), nullable=False)
    sort = Column(Integer, server_default="0")
    status = Column(Integer, server_default="1")
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
